Set some proper paths to make cmake find our tools.
Original patch by Heiko Przybyl

--- cmake-2.6.4/Modules/Platform/Darwin.cmake
+++ cmake-2.6.4/Modules/Platform/Darwin.cmake
@@ -141,18 +141,22 @@
 
  
 # default to searching for frameworks first
-SET(CMAKE_FIND_FRAMEWORK FIRST)
+SET(CMAKE_FIND_FRAMEWORK LAST)
 # set up the default search directories for frameworks
 SET(CMAKE_SYSTEM_FRAMEWORK_PATH
+  @GENTOO_PORTAGE_EPREFIX@/Frameworks
+  @GENTOO_PORTAGE_EPREFIX@/usr/lib
   ~/Library/Frameworks
   /Library/Frameworks
   /Network/Library/Frameworks
   /System/Library/Frameworks)
 
 # default to searching for application bundles first
-SET(CMAKE_FIND_APPBUNDLE FIRST)
+SET(CMAKE_FIND_APPBUNDLE LAST)
 # set up the default search directories for application bundles
 SET(CMAKE_SYSTEM_APPBUNDLE_PATH
+  @GENTOO_PORTAGE_EPREFIX@/Applications
+  @GENTOO_PORTAGE_EPREFIX@/usr/bin
   ~/Applications
   /Applications
   /Developer/Applications)
--- cmake-2.6.4/Modules/Platform/UnixPaths.cmake
+++ cmake-2.6.4/Modules/Platform/UnixPaths.cmake
@@ -11,6 +11,7 @@
   # Standard
   # / /usr /usr/local
   "${CMAKE_INSTALL_PREFIX}" "${CMAKE_INSTALL_PREFIX}/.."
+  @GENTOO_PORTAGE_EPREFIX@ @GENTOO_PORTAGE_EPREFIX@/usr
   / /usr
 
   # CMake install location
@@ -22,38 +23,33 @@
 
 # List common include file locations not under the common prefixes.
 LIST(APPEND CMAKE_SYSTEM_INCLUDE_PATH
-  # Windows API on Cygwin
-  # /usr/include/w32api
-
-  # X11
-  # /usr/X11R6/include /usr/include/X11
-
-  # Other
-  # /opt/local/include /usr/pkg/include
-  # /opt/csw/include /opt/include  
-  # /usr/openwin/include
+	@GENTOO_PORTAGE_EPREFIX@/usr/include
   )
 
 LIST(APPEND CMAKE_SYSTEM_LIBRARY_PATH
-  # Windows API on Cygwin
-  # /usr/lib/w32api
-
-  # X11
-  # /usr/X11R6/lib /usr/lib/X11
-
-  # Other
-  # /opt/local/lib /usr/pkg/lib
-  # /opt/csw/lib /opt/lib 
-  # /usr/openwin/lib
+	@GENTOO_PORTAGE_GCCLIBDIR@/gcc
+	@GENTOO_PORTAGE_GCCLIBDIR@
+	@GENTOO_PORTAGE_EPREFIX@/usr/lib64
+	@GENTOO_PORTAGE_EPREFIX@/usr/lib32
+	@GENTOO_PORTAGE_EPREFIX@/usr/lib
+	@GENTOO_PORTAGE_EPREFIX@/lib
   )
 
 LIST(APPEND CMAKE_SYSTEM_PROGRAM_PATH
   # /usr/pkg/bin
   "${CMAKE_INSTALL_PREFIX}/bin"
+  @GENTOO_PORTAGE_EPREFIX@/usr/bin
+  @GENTOO_PORTAGE_EPREFIX@/bin
   )
 
 LIST(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
   "${CMAKE_INSTALL_PREFIX}/lib" "${CMAKE_INSTALL_PREFIX}/../lib"
+	@GENTOO_PORTAGE_GCCLIBDIR@/gcc
+	@GENTOO_PORTAGE_GCCLIBDIR@
+	@GENTOO_PORTAGE_EPREFIX@/usr/lib64
+	@GENTOO_PORTAGE_EPREFIX@/usr/lib32
+	@GENTOO_PORTAGE_EPREFIX@/usr/lib
+	@GENTOO_PORTAGE_EPREFIX@/lib
   /lib /usr/lib /usr/lib32 /usr/lib64
   )
 
